src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -215,37 +214,86 @@ private static void ChooseBestBlobTransactions(
             in UInt256 baseFee,
             ArrayPoolList<Transaction> selectedBlobTxs)
         {
-            int size = leftoverCapacity + 1;
+            int maxCapacity = leftoverCapacity + 1;
             // The maximum total fee achievable with capacity
-            using ArrayPoolList<ulong> dpFeesPooled = new(capacity: size, count: size);
+            using ArrayPoolList<ulong> dpFeesPooled = new(capacity: maxCapacity, count: maxCapacity);
             Span<ulong> dpFees = dpFeesPooled.AsSpan();
 
-            using ArrayPoolBitMap isChosen = new(candidateTxs.Count * size);
+            using ArrayPoolBitMap isChosen = new(candidateTxs.Count * maxCapacity);
 
             // Build up the DP table to find the maximum total fee for each capacity.
             // Outer loop: go through each transaction (1-based index).
             // Inner loop: iterate capacity in descending order to avoid overwriting data needed for the calculation.
             for (int i = 0; i < candidateTxs.Count; i++)
             {
                 Transaction tx = candidateTxs[i];
+
                 if (!tx.TryCalculatePremiumPerGas(baseFee, out UInt256 premiumPerGas))
                 {
+                    // Skip any tx where where tx can't cover the premium per gas.
                     continue;
                 }
+
+                // How many blobs does this tx actually consume?
                 int blobCount = tx.GetBlobCount();
-                // Use actual gas used when available as the tx may be using over-estimated gaslimit
-                ulong feeValue = (ulong)premiumPerGas * (ulong)(tx.SpentGas);
+                // If this tx has explicit dependencies (i.e. it requires k prior blobs
+                // from the *same address* to be in the block before it), include them here.
+                // We'll need a capacity of blobDependenciesCount slots *plus* its own blobCount.
+                int blobCapacityNeeded = tx.BlobDependenciesCount + blobCount;
```

Comment:
Can be a dictionary<tx, int>, uglier but without adding tx field needed only here

Reply:
Problematic when testing tx pools with 250k txs? That dictionary will be massive

Reply:
Actually could return with enumerator; as pair though would be more ugly 🤔

Reply:
Removed the extra property via two sort methods